src/Integration/LocalServices/ProjectSystemHelper.cs

```diff
@@ -502,5 +492,47 @@ public bool IsLegacyProjectSystem(Project dteProject)
 
             return hierarchy != null && hierarchy is IVsAggregatableProjectCorrected;
         }
+
+        public IEnumerable<VSConstants.VSITEMID> GetAllItems(IVsHierarchy vsHierarchy)
+        {
+            // Based on https://github.com/microsoft/VSSDK-Extensibility-Samples/blob/c6b80a4ff7023578649d7edecc8fd6cd8a34da10/Code_Sweep/C%23/VsPackage/BuildManager.cs#L450
+            return ChildrenOf(vsHierarchy, (uint) VSConstants.VSITEMID.Root);
+        }
+
+        private static IEnumerable<VSConstants.VSITEMID> ChildrenOf(IVsHierarchy hierarchy, uint rootId)
+        {
+            var result = new List<VSConstants.VSITEMID>();
+
+            for (var itemId = FirstChild(hierarchy, rootId);
+                itemId != (uint)VSConstants.VSITEMID.Nil;
+                itemId = NextSibling(hierarchy, itemId))
+            {
+                result.Add((VSConstants.VSITEMID) itemId);
```

Comment:
Did you decide against making this enumeration lazy?

Reply:
Ah - so the top-level projects are enumerated lazily, but all of the items in a project are fetched at once?
It's worth documenting this as it's not obvious.

Reply:
I added a comment 👍